packages/block-editor/src/components/block-list/block.js

```diff
@@ -511,22 +527,18 @@ function BlockListBlock( {
 				rootClientId={ rootClientId }
 			/>
 			{ isFirstMultiSelected && (
-				<BlockMultiControls rootClientId={ rootClientId } />
+				<BlockMultiControls
+					rootClientId={ rootClientId }
+					moverOptions={ moverOptions }
+				/>
 			) }
-			<div className="editor-block-list__block-edit block-editor-block-list__block-edit">
-				{ shouldRenderMovers && (
-					<BlockMover
-						clientIds={ clientId }
-						blockElementId={ blockElementId }
-						isHidden={ ! isSelected }
-						isDraggable={
-							isDraggable !== false &&
-							( ! isPartOfMultiSelection && isMovable )
-						}
-						onDragStart={ onDragStart }
-						onDragEnd={ onDragEnd }
-					/>
+			<div
+				className={ classnames(
+					'editor-block-list__block-edit block-editor-block-list__block-edit',
+					{ 'has-mover-inside': position === 'inside' },
 				) }
+			>
+				{ shouldRenderMovers && ( position === 'outside' ) && blockMover }
```

Comment:
Do we really need the outside property? Can't we always render the movers inside or outside and then with CSS style them to look the way we want (inside or outside)?
If not no can we for now alway render horizontal movers inside and vertical ones outside?

The reason I would prefer to avoid inside or outside is that it seems a big contract that forces us to render the mover in a specific position. What if later we redesign the movers and all of them only make sense inside the block area? What if we move them to the toolbar?

Reply:
rendering in one place is not a feasible solution as it would open a host of new issues, the most basic one is breaking the tab order, and ending with weirdly complex CSS for positioning.

I would better go for the second option and simply remove the 'position', however these two settings allow for great flexibility if we keep the current concept of movers. To answer your question, if we move the movers to the toolbar we simply deprecate the 'position' option, although we may find that we don't need any option in that case.

Reply:
I'm wondering if we need this much flexibility? When would we use `position=outside, orientation=horizontal` or `position=inside, orientation=vertical`?

Reply:
That's true. The positional use case that I know would be for specifying whether the mover is before the BlockEdit or after. In the gallery it'd be before, and in the nav item, after.

I think it'd be good to consider how that might be added later without actually doing it in this PR.

Reply:
The problem is that if we simply ignore a position property and just render differently based on `orientation` alone we make assumptions about what horizontal means for block implementers. 

Intent is highly valuable and I do believe that by default people just want horizontal movers. However the fact that simply saying:

```
<InnerBlocks moverStyle='horizontal' /> 
```

causes the rendering of the HTML to change DOM position is something you do not expect. How could one assume that?

Offering both options is an alarm bell that tells you there is this option too to do this:

```
<InnerBlocks moverStyle='horizontal' moverPosition='beforeBlock' /> 
<InnerBlocks moverStyle='vertical' moverPosition='afterBlock' /> 
```

I will update the code to reflect this approach.

Reply:
Some of the technical aspects are a litte above me here, but it feels like we are adding a lot of options and optimizing for something that may very probably change in the future. What if we, for example, decide that the horizontal movers exist only in the block toolbar?

Another reason to be very careful in introducing options for block developers, is that we don't want block implementers to turn these controls off, as they serve important accessibility functions. 

Take the slot fill names, for example — we have intentionally kept these named after their _purpose_, not after their _location_. That way as we continue to iterate the block UI, if we have to move a slot-fill location, we don't have to rename it, and plugins will still work.

Reply:
> Take the slot fill names, for example — we have intentionally kept these named after their purpose, not after their location.

Well put! This is why I'd prefer something like `<InnerBlocks moverStyle="horizontal" />`. Where the movers appear in the DOM is an implementation detail that block authors needn't care about.

Reply:
> Where the movers appear in the DOM is an implementation detail that block authors needn't care about.

I don't agree with that. Being able to render it in the right place for a correct tab order is a level of control that block authors need to have. `<InnerBlocks moverStyle="horizontal" />` just won't cut it for a lot of use cases.

That said, I think the right thing to do is pick one approach, mark it as experimental, and then not remove that experimental prefix until we're tried implementing a few different blocks with these movers.